programs/validator-history/src/state.rs

```diff
@@ -184,8 +184,53 @@ impl CircBuf {
         &mut self.arr
     }
 
-    /// Returns &ValidatorHistoryEntry for each existing entry in range [start_epoch, end_epoch], factoring for wraparound
-    /// Returns None if either start_epoch or end_epoch is not in the CircBuf
+    /// Given a new entry and epoch, inserts the entry into the buffer in sorted order
+    /// Will not insert if the epoch is out of range or already exists in the buffer
+    fn insert(&mut self, entry: ValidatorHistoryEntry, epoch: u16) -> Result<()> {
```

Comment:
rip, not doing index = epoch % self.history.len() might have been a mistake

Reply:
💯